### PR TITLE
Rails4 search

### PR DESCRIPTION
**Pull request by yanyangfeng**
Basically all the changes you saw at the meeting today (6/27/17), with the addition of a few bug fixes in handling search results with no interactions. 

To summarize:
- Drug/Gene/Interactions view updates
- Search form updates (preset filters, facelift)
- Search results updates (sidebar with toggling, ambiguous vs definite matches)
- Various visual fixes to site

### CODE REVIEW

**Comment by ahwagner** [comment]
Hey @yanyangfeng, it looks like there's a test failure here associated with filter chaining that needs to be addressed.

**Comment by susannasiebert** [comment]
CI, test this.

File: app/filters/clinically_actionable.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  end
+
+  def axis
+    :clinically_actionable
+  end
+
+  def resolve
+    Set.new DataModel::Gene
+      .joins(:gene_categories)
+      .where("gene_categories_genes.gene_claim_category_id = ?", '0d157beb-fd17-404d-8755-3a81aa5ed704')
```
Comment (susannasiebert) [inline app/filters/clinically_actionable.rb]: Do we know if this ID might change in the future? Should we filter on the category name instead?

File: app/controllers/interaction_claims_controller.rb [hunk 2]
```diff
@@ -49,5 +55,8 @@ def perform_interaction_search
     validate_interaction_request(params)
     search_results = LookupInteractions.find(params)
     @search_results = InteractionSearchResultsPresenter.new(search_results, start_time, view_context)
+    puts "YOOOOOOO"
```
Comment (susannasiebert) [inline app/controllers/interaction_claims_controller.rb]: Some stray `puts` that can be removed.

File: app/filters/druggable_genome.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  end
+
+  def axis
+    :druggable_genome
+  end
+
+  def resolve
+    Set.new DataModel::Gene
+      .joins(:gene_categories)
+      .where("gene_categories_genes.gene_claim_category_id = ?", 'd3ec2631e0b2434b9dcc008e793d3fa5')
```
Comment (susannasiebert) [inline app/filters/druggable_genome.rb]: Same as above.

File: app/filters/drug_resistance.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  end
+
+  def axis
+    :drug_resistance
+  end
+
+  def resolve
+    Set.new DataModel::Gene
+      .joins(:gene_categories)
+      .where("gene_categories_genes.gene_claim_category_id = ?", '655bedfa9d71482796a5a34b87dfb297')
```
Comment (susannasiebert) [inline app/filters/drug_resistance.rb]: Same as above.

File: app/models/interaction_search_result.rb [hunk 5]
```diff
@@ -1,25 +1,23 @@
 class InteractionSearchResult
```
Comment (susannasiebert) [inline app/models/interaction_search_result.rb]: This class was also changed in #190 so we might see some merge conflicts here depending on who merges first.

**Comment by ahwagner** [comment]
closes #174

**Pull request merged**